python/mxnet/ndarray/sparse.py

```diff
@@ -1159,6 +1185,293 @@ def _ndarray_cls(handle, writable=True, stype=_STORAGE_TYPE_UNDEFINED):
 _set_ndarray_class(_ndarray_cls)
 
 
+def add(lhs, rhs):
+    """Returns element-wise sum of the input arrays with broadcasting.
+
+    Equivalent to ``lhs + rhs``, ``mx.nd.broadcast_add(lhs, rhs)`` and
+    ``mx.nd.broadcast_plus(lhs, rhs)`` when shapes of lhs and rhs do not
+    match. If lhs.shape == rhs.shape, this is equivalent to
+    ``mx.nd.elemwise_add(lhs, rhs)``
+
+    .. note::
+
+        If the corresponding dimensions of two arrays have the same size or one of them has size 1,
+        then the arrays are broadcastable to a common shape.abs
+
+    Parameters
+    ----------
+    lhs : scalar or array
+        First array to be added.
+    rhs : scalar or array
+         Second array to be added.
+        If ``lhs.shape != rhs.shape``, they must be
+        broadcastable to a common shape.
+
+    Returns
+    -------
+    NDArray
+        The element-wise sum of the input arrays.
+
+    Examples
+    --------
+    >>> x = mx.nd.ones((2,3))
```

Comment:
I think we should replace the example with the supported case specifically for sparse:

```
>>> x = mx.nd.ones((2,3)).tostype('csr')
>>> y = mx.nd.ones((2,3)).tostype('csr')
>>> (x + y).asnumpy()
    array([[ 2.,  2.,  2.],
            [ 2.,  2.,  2.]], dtype=float32)
>>> (x + 2).asnumpy()
    array([[ 3.,  3.,  3.],
            [ 3.,  3.,  3.]], dtype=float32)
>>> mx.nd.sparse.add(x,y).asnumpy()
    array([[ 2.,  2.,  2.],
            [ 2.,  2.,  2.]], dtype=float32)

```